src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/AddressType.php

```diff
@@ -117,17 +119,32 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                     $form->add('customer', CustomerCheckoutGuestType::class, ['constraints' => [new Valid()]]);
                 }
             })
-            ->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event): void {
+            ->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) use ($options): void {
                 $orderData = $event->getData();
 
+                /** @var ChannelAwareInterface $order */
+                $order = $options['data'];
+                /** @var ChannelInterface $channel */
+                $channel = $order->getChannel();
+
                 $differentBillingAddress = $orderData['differentBillingAddress'] ?? false;
                 $differentShippingAddress = $orderData['differentShippingAddress'] ?? false;
 
-                if (isset($orderData['billingAddress']) && !$differentBillingAddress && !$differentShippingAddress) {
+                if (
+                    isset($orderData['billingAddress']) &&
+                    !$differentBillingAddress &&
+                    !$differentShippingAddress &&
+                    !$channel->isShippingAddressInCheckoutRequired()
```

Comment:
**A lot** of conditions :D maybe it would be worth to make a checker private function with them for better readability (similar below)